API/Controllers/TournamentsController.cs

```diff
@@ -1,34 +1,223 @@
+    private readonly IMatchesService _matchesService = matchesService;
 
+    /// <summary>
+    /// List all tournaments
+    /// </summary>
+    /// <remarks>Will not include match data</remarks>
+    /// <response code="200">Returns all tournaments</response>
     [HttpGet]
-    [Authorize(Roles = "system")]
-    public async Task<IActionResult> GetAsync()
+    [Authorize(Roles = "admin, system")]
+    [ProducesResponseType<IEnumerable<TournamentDTO>>(StatusCodes.Status200OK)]
+    public async Task<IActionResult> ListAsync()
     {
-        IEnumerable<TournamentDTO> res = await _service.GetAllAsync();
-        return Ok(res);
+        IEnumerable<TournamentDTO> result = await _tournamentsService.ListAsync();
+        return Ok(result);
     }
 
-    [HttpPut]
-    [Authorize(Roles = "admin, system")]
-    public async Task<IActionResult> PutAsync(TournamentWebSubmissionDTO tournament)
+    /// <summary>
+    /// Submit a tournament
+    /// </summary>
+    /// <remarks>
+    /// The user submits a tournament (containing metadata and links) to the front-end. They are looking to add new data
+    /// to our database that will eventually count towards ratings.
+    ///
+    /// This post endpoint takes these links, validates them (i.e. checks for duplicates,
+    /// whether the match titles align with osu! tournament naming conventions,
+    /// amount of matches being submitted, etc.).
+    ///
+    /// Assuming we have a good batch, we will mark all of the new items as "PENDING".
+    /// The API.Osu.Multiplayer.MultiplayerLobbyDataWorker service checks the database for pending links
+    /// periodically and processes them automatically.
+    /// </remarks>
+    /// <param name="wrapper">Tournament submission data</param>
+    /// <param name="verify">Optionally verify all included matches</param>
+    /// <response code="401">If verify is true and the User does not have match verification privileges</response>
+    /// <response code="400">If a tournament matching the given name and mode already exists, or the given <see cref="wrapper"/> is malformed</response>
+    /// <response code="201">Returns location information for the created tournament</response>
+    [HttpPost]
+    [ProducesResponseType<ModelStateDictionary>(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType<TournamentCreatedResultDTO>(StatusCodes.Status201Created)]
+    public async Task<IActionResult> CreateAsync(
+        [FromBody] TournamentWebSubmissionDTO wrapper,
+        [FromQuery] bool verify = false
+    )
+    {
+        if (!ModelState.IsValid)
+        {
+            return BadRequest(ModelState);
+        }
+
+        // Determine verification source
+        MatchVerificationSource? verificationSource = null;
+        if (verify)
+        {
+            if (!User.CanVerifyMatches())
+            {
+                return Unauthorized();
+            }
```

Comment:
I think these if statements can be combined. Then `verificationSource` can be created & assigned in one line.